Set Up Components - Milestone 2

AbdumurodovaZulfizar opened the pull request:
# Milestone 2 - Components

### What I have done on this milestone:

- Created four components - App, Display, Button, ButtonPanel
- Implement design on these components

_Done by Zulfizar Abdumurodova(Full-Stack software developer)_

_Special thanks to all code reviewers_

addod19 commented on src/components/App.js:
```diff
@@ -4,8 +4,8 @@ import ButtonPanel from './ButtonPanel';
   return (
-    <div className="App">
-      <Display />
+    <div className="App my-5">
```
- [ ] It would be better if we wrap the children components with the `<React.Fragment>` or the short syntax `<>`

AbdumurodovaZulfizar replied:
But I have used this div with the class of App to centre the div.

addod19 commented on src/components/ButtonPanel.js:
```diff
@@ -1,7 +1,50 @@
+import React from 'react';
+import Button from './buttons/button';
+
+/* eslint-disable */
```
- [ ] It would be better if we fix the ESLint errors instead of disabling the rules. This is a standard for all students of Microverse, it helps us write clean and structured code

AbdumurodovaZulfizar replied:
Done 💯

addod19 commented on src/components/Display.js:
```diff
@@ -1,7 +1,22 @@
+
+/* eslint-disable*/
+
+Display.propTypes = {
```
- [ ] It would be better if the resulting prop has a default value of `0`, in our case `previousVal` or `currentVal`

AbdumurodovaZulfizar replied:
![image](https://user-images.githubusercontent.com/77872355/133628970-0acffaf0-2d14-46d2-9229-1a832798183a.png)
It is 0 by default, and I have used properties, so I will use them to change in the future milestones

addod19 commented on src/components/buttons/button.js:
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+/*eslint-disable*/
```
- [x]  It would be better if we fix the ESLint errors instead of disabling the rules. This is a standard for all students of Microverse, it helps us write clean and structured code

AbdumurodovaZulfizar replied:
Done 💯

addod19 commented on src/components/ButtonPanel.js:
```diff
@@ -1,7 +1,50 @@
+import Button from './buttons/button';
+
+/* eslint-disable */
+
```
- [ ] It would be better if we group the elements in the following order:
![Screenshot from 2021-09-16 14-04-02](https://user-images.githubusercontent.com/46388113/133626820-dd8f9b82-8514-437a-b2ed-ee3e6a81a382.png)

AbdumurodovaZulfizar replied:
![image](https://user-images.githubusercontent.com/77872355/133629212-b64e2901-8f3e-41c3-94b4-22428a7f2248.png)
It is that like order on the screen, I have grouped them on arrays to place them on that like groups

addod19 reviewed (changes requested):
### Hi @AbdumurodovaZulfizar, 

Good job so far! Git-Flow followed :fireworks: README and commis message professional and descriptive :smile: Most of the components implemented as expected :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

__Happy coding__ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

afanimartin commented on src/components/App.js:
```diff
@@ -4,8 +4,8 @@ import ButtonPanel from './ButtonPanel';
   return (
-    <div className="App">
-      <Display />
+    <div className="App my-5">
```
Please use the React.Fragment short syntax as suggested.

AbdumurodovaZulfizar replied:
As you can see I have added class to this, I have added fragments to the Display component, but on this occasion, if I remove this div, I couldn't add class, I ask to approve my project.

afanimartin reviewed (changes requested):
Hi @AbdumurodovaZulfizar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AbdumurodovaZulfizar commented:
Although it was not my fault for the previously required change, I have changed the code structure. Why I am saying this is that the main aim of implementing React.Fragment is avoiding extra nodes, but on this occasion I have removed div, but added it to the HTML file

rukundoeric commented on src/components/App.js:
```diff
@@ -4,10 +4,10 @@ import ButtonPanel from './ButtonPanel';
-    <div className="App">
-      <Display />
+    <>
+      <Display previousVal="" currentVal="0" />
```
- [ ] <b>OPTIONAL:</b> You have already set default values in `Display` component in `Display.defaultProps`, that means, you don't have to do it here. since this wasn't mentioned in the previous review, you can consider it as optional, but I strongly recommend you to fix it.

rukundoeric commented on src/components/ButtonPanel.js:
```diff
@@ -1,7 +1,50 @@
+import React from 'react';
+import Button from './buttons/button';
+
+/* eslint-disable */
```
- [ ] <b>REQUIRED:</b> You are not allowed to use `eslint-disable` unless it is totally obvious that you have to. it's like making changes in Eslint config. they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/) and [Zendesk](https://microverse.zendesk.com/hc/en-us/articles/1500002085202-February-2nd-2021-linter-errors-as-a-reason-of-project-invalidation-)). kindy try to fix the errors instead of ignoring them.

rukundoeric reviewed (changes requested):
Hi @AbdumurodovaZulfizar, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

rukundoeric commented on src/components/ButtonPanel.js:
```diff
@@ -1,7 +1,49 @@
+import React from 'react';
+import Button from './buttons/button';
+
+// /* eslint-disable */
```
- [ ] <b>REQUIRED:</b> Actually, this doesn't make any difference, `eslint-disable` keywords are designed to be comments. it doesn't matter what kind of comment is used as long as it is there. So kindly please remove it.

AbdumurodovaZulfizar replied:
It was not working actually

rukundoeric reviewed (changes requested):
Hi @AbdumurodovaZulfizar, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

rindrajosia reviewed (approved):
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
